Magmap release v1.0.0 (#77)
nf-core/magmap: docs/usage.md

TREATMENT_REP3,AEG588A6_S6_L004_R1_001.fastq.gz,
```

| Column    | Description                                                                                                                                                                            |
| --------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `sample`  | Custom sample name. This entry will be identical for multiple sequencing libraries/runs from the same sample. Spaces in sample names are automatically converted to underscores (`_`). |
| `fastq_1` | Full path to FastQ file for Illumina short reads 1. File has to be gzipped and have the extension ".fastq.gz" or ".fq.gz".                                                             |
| `fastq_2` | Full path to FastQ file for Illumina short reads 2. File has to be gzipped and have the extension ".fastq.gz" or ".fq.gz".                                                             |

An [example samplesheet](../assets/samplesheet.csv) has been provided with the pipeline.

### Genomes input

> Magmap needs two mandatory input to run. One, the samplesheet, was explained above while the second, genome input needs to be specified with the option `--genomeinfo`. The file is a `.csv` file and it requires three columns: accno, genome_fna, genome_gff. it looks as follow:

Reviewer: ```suggestion
The second mandatory input is the genome input sheet, which needs to be specified with the option `--genomeinfo`. The file is a `.csv` file and it requires three columns: accno, genome_fna, genome_gff. it looks as follow:
```

```csv title="samplesheet.csv"
accno,genome_fna,genome_gff
GCA_002688505,./genomes/GCA_002688505.fna,./genomes/GCA_002688505.gff
GCA_002688515,./genomes/GCA_002688515.fna,
```

> 

Reviewer: Missing table describing what goes in each column here, as you had above for the samplesheet

N.B.: you don't need to provide gff, Prokka can handle if you only provide fasta files but we recommend when adding gff that the file is generated by using prokka in order to don't get any conflict with other generated gffs.

### Other inputs (optional)

Magmap can handle several types of input that can be used for different purpose.

#### Indexes input

> The indexes input is used by Sourmash to select genomes that can downloaded in a second step and added to the pipeline.

Reviewer: ```suggestion
The indexes input is used by Sourmash to select genomes that can be downloaded in a second step and added to the pipeline.
```

It is provided with the `--indexes` parameter and it is a path (local or remote).

```
https://github.com/nf-core/test-datasets/raw/magmap/testdata/sourmash_test.index.sbt.zip

> ```

Reviewer: What is this? Is this meant t obe an example? If so add the parameter before the URL

Reviewer: And don't forget to specify the code block langauge/syntax

N.B.: the sbt files can be generated with sourmash (check it [here](https://sourmash.readthedocs.io/en/latest/index.html)) or some can be found in sourmash documentation website.
E.g. "sourmash gtdb sbt" you will find the indexes for gtdb.

> #### metadata input

Reviewer: ```suggestion
#### Metadata input
```

Magmap accepts several metadata as `.csv` files that provides information about the genomes that you will use in the pipeline.

##### gtdb_metadata

> This file contains several information that can be found in gtdb metadata on the official [website](https://data.ace.uq.edu.au/public/gtdb/data/releases/release80/80.0/). You can either use directly the same file found there or you can make a custom one.

Reviewer: What file is this exactly? There are multiple files at that link (and that link is for a very old version of GTDB! It might not work with the latest version of the tool if you're using a recent module)

Author: updated!

##### gtdb-tk metadata

> This file contains several information that can be obtained as gtdb-tk output. You can either use directly the same file found there or you can make a custom one.

Reviewer: For this and all the other optional inputs, please describe exactly what this file should contain/look like, this is very unclear to me know and as I a user  I would have 0 idea why I would use it or how to create it.

##### checkM metadata

This file contains several information that can be obtained as checkM output. You can either use directly the same file found there or you can make a custom one.

### Check duplicates

> The pipeline is checking if there are any duplicate names among the genomes that the user provides. If there are duplicates, the pipeline will stop and return a file with the contig names that needs to be changed in their name in order to work. This is done to avoid overlapping in the following steps (e.g. same prokka output for the protein sequences and the gffs).

Reviewer: ```suggestion
The pipeline will perform validation checks to see if there are any duplicate names among the genomes that the user provides. If there are duplicates, the pipeline will stop and return a file with the contig names that needs to be changed in their name in order to work. This is done to avoid overlapping in the following steps (e.g. same prokka output for the protein sequences and the gffs).
```

### Filter/remove sequences from the samples (e.g. rRNA sequences with SILVA database)

The pipeline can remove potential contaminants using the BBduk program.

> Specify a fasta file, gzipped or not, with the --sequence_filter sequences.fasta parameter.

Reviewer: ```suggestion
Specify a fasta file, gzipped or not, with the `--sequence_filter <sequences>.fasta` parameter.
```

Make sure to wrap short code examples in code blocks to make it easier to recognise

For further documentation, see the [BBduk official website](https://jgi.doe.gov/data-and-tools/software-tools/bbtools/bb-tools-user-guide/bbduk-guide/).

```bash
nextflow run nf-core/magmap -profile docker --outdir results/ --input samples.csv --genomeinfo localgenomes.csv --sequence_filter path/to/file
```

### Sourmash (optional)

> with [Sourmash](https://sourmash.readthedocs.io/en/latest/index.html) you can filter the genomes to be used by magmap in the mapping step. This function is optional but can speed up the process and let you get a better genomes /reads mapping ratio since you are removing all the genomes that are not passing the threshold (that you can select).

Reviewer: ```suggestion
With [Sourmash](https://sourmash.readthedocs.io/en/latest/index.html) you can filter the genomes to be used by magmap in the mapping step. This function is optional but can speed up the process and let you get a better genomes /reads mapping ratio since you are removing all the genomes that are not passing the threshold (that you can select).
```

What threshold? Give an example parameter in the command

```bash
nextflow run nf-core/magmap -profile docker --outdir results/ --input samples.csv --genomeinfo localgenomes.csv --sourmash true
```

### ORF caller option

> The pipeline uses [Prokka](https://github.com/tseemann/prokka) to call genes/ORFs from the genomes. This is suitable for prokaryotes and it provides a gff as output for downstream analysis. It also performs functional annotation of ORFs.

Reviewer: Why do you need this for mapping?

Author: it's actually used with FeatureCounts. For that program we need the gffs.
